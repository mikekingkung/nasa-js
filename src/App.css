* {
  box-sizing: border-box;
  margin: 10 px;
  padding: 10 px;
  background: rgb(0, 0, 0);
  color: rgb(156, 20, 43);  
}

body {
  font-family: Arial, Helvetica, sans-serif;
  line-height: 1.6;
}

.container {
  margin: 10px auto;
  background: rgb(0, 0 ,0);
  color: rgb(156, 20, 43);
}

input[type="text"] {
  margin: 5px 0;
  padding: 5px;
  width: 100%;
  font-size: 1.1rem;
}

input[type="submit"] {
  width: 100%;
  background: rgb(165, 209, 82);
  color: rgb(156, 20, 43);
  padding: 5px;
  font-size: 1.1rem;
  cursor: pointer;
}
