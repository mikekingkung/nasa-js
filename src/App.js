import './App.css';
import Header from './components/header/Header';
import Main from './components/main/Main';
import Footer from './components/footer/Footer';
import { BrowserRouter as Router} from 'react-router-dom';
import MyContext from './MyContext';

function App() {
  return (
    <>
      <MyContext.Provider value='Guest'>
      <Router>
        <Header />
        <Main />
        <Footer />
      </Router>
      </MyContext.Provider>
    </>
  );
}

export default App;
