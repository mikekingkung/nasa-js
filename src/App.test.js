import { render, screen } from '@testing-library/react';
import App from './App';

describe('<App component', () => {
  it('Correctly renders' ,() => {
      const wrapper = render(<App/>);
      expect(wrapper).toMatchSnapshot();
  });
});


