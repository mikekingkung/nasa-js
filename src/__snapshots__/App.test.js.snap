// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App component Correctly renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="container"
        >
          <nav
            class="navbar navbar-expand-lg navbar-dark"
          >
            <div
              class="container-fluid"
            >
              <a
                class="navbar-brand"
                href="/"
              >
                <img
                  alt="Learn Space Ltd"
                  src="space.jpg"
                />
              </a>
              <div
                class="collapse navbar-collapse"
                id="navbarSupportedContent"
              >
                <ul
                  class="navbar-nav"
                >
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/discover"
                    >
                      Discover
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/topics"
                    >
                      Topics
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/links"
                    >
                      Links
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/about"
                    >
                      About
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/secured"
                    >
                      Secured
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/signup"
                    >
                      Sign Up
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/userdetails"
                    >
                      User Details
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/form"
                    >
                      Form
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
          <img
            alt="Learn Space Ltd"
            class="img-fluid"
            src="bannerimage.png"
          />
        </div>
      </header>
      <main
        class="container"
      >
        <div>
          <p
            data-testid="welcome"
          >
            Welcome to the Learn Space application. The application makes use fof the Nasa Video and Image API. The application allows keyword searches. Results are limited ae 10 currently per search. It is also possible to save links and review them on the user details page.
          </p>
          <img
            alt="Learn Space Ltd"
            class="img-fluid"
            src="landingPageImage.jpg"
          />
        </div>
      </main>
      <div>
        <img
          alt="footer"
          src="footerimage.png"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="container"
      >
        <nav
          class="navbar navbar-expand-lg navbar-dark"
        >
          <div
            class="container-fluid"
          >
            <a
              class="navbar-brand"
              href="/"
            >
              <img
                alt="Learn Space Ltd"
                src="space.jpg"
              />
            </a>
            <div
              class="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul
                class="navbar-nav"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/discover"
                  >
                    Discover
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/topics"
                  >
                    Topics
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/links"
                  >
                    Links
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/about"
                  >
                    About
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/secured"
                  >
                    Secured
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/signup"
                  >
                    Sign Up
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/userdetails"
                  >
                    User Details
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/form"
                  >
                    Form
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
        <img
          alt="Learn Space Ltd"
          class="img-fluid"
          src="bannerimage.png"
        />
      </div>
    </header>
    <main
      class="container"
    >
      <div>
        <p
          data-testid="welcome"
        >
          Welcome to the Learn Space application. The application makes use fof the Nasa Video and Image API. The application allows keyword searches. Results are limited ae 10 currently per search. It is also possible to save links and review them on the user details page.
        </p>
        <img
          alt="Learn Space Ltd"
          class="img-fluid"
          src="landingPageImage.jpg"
        />
      </div>
    </main>
    <div>
      <img
        alt="footer"
        src="footerimage.png"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
